code/kobe.py

```diff
@@ -1107,32 +1117,543 @@ def kobe_shadows(df_input,cdpp):
+    
+    ## 0 - create arrays to go through each bin       
+    ## 1 - sanity checks
+    drop_c=0
+    drop1_c=0
+    
+    ## Create Columns 
+    ## 1 create columns and mark each as '1'. This ensures that all potential planets with MES>50 are marked TCEs.
+    df_kobe_output['flag_det_eff'] = 1
+    ## then all planets that are not tce are marked with -1
+    
+    ## select bin width, the MES threshold after which potential planets are marked TCEs
+    min_bin = 0
+    max_bin = 50
+    bin_wid = 0.5
+    n_bins = ((max_bin - min_bin)/bin_wid)
+    
+    for i in np.arange(min_bin, max_bin, bin_wid):
+        min_mes = i
+        max_mes = min_mes + 0.5
+        
+        #select all planets that fall within the bin (min_mes, max_mes)
+        df_bin = df_kobe_output.loc[df_kobe_output['kobe_mes']>=min_mes].loc[df_kobe_output['kobe_mes']<max_mes]    
+        
+        for mes in df_bin['kobe_mes']:
+        ## 2 now we are inside a bin
+        ## Gamma CDF calculation and coefficients taken directly from Christiansen 2017
+            
+            a = 30.87
+            b = 0.271
+            c = 0.94
+            mid_bin = (np.amax(df_bin['kobe_mes'])+np.amin(df_bin['kobe_mes']))/2
+            detection_eff = c*stats.gamma.cdf(mid_bin, a, scale = b)
+            
+        drop_ratio_c = 1 - detection_eff
+            
+        ## all planets that fall into current bin are selected
+
+        condition = (df_kobe_output['kobe_mes']>=min_mes) & (df_kobe_output['kobe_mes']< max_mes)
+        planet_index = df_kobe_output.loc[condition].index.to_numpy()
+
+        ## estimate number of planets to flag
+        drop_planets_size_c = int(np.floor(drop_ratio_c*planet_index.shape[0]))
+        drop_c = drop_c + drop_planets_size_c
+
+        
+        ## now we randomly select planets within this bin which are not flagged as TCEs
+        if planet_index.shape[0] != 0:
+            drop_planets_index_c = list(np.random.choice(planet_index,size=drop_planets_size_c,replace=False))
+            df_kobe_output.loc[drop_planets_index_c,'flag_det_eff'] = -1
+            drop1_c = drop1_c + len(drop_planets_index_c)
+
+    if print_details == True:
+        print('----------------------------------')
+        print('For identifying consistent TCEs, bins distributed between MES value of %d and %d'%(min_bin, max_bin))
+        print('Number of bins along given MES range: ', n_bins) 
+        print('Detection Efficiency Calculation Information: \nDetection efficiency ranges from 0-1 \nDetection efficiency flagged as -1 when not a TCE \nSpacing is %.1f \n'%(bin_wid))
+        print('----------------------------------')
+    
+    return df_kobe_output
+
+
 def kobe_transits(df_kobe_output):
     """
     The KOBE Transits module: calculate transit related parameters for all planets
 
     Parameters
     ----------
-    df_input : pandas dataframe
-        input dataframe processed suitably
-
+    df_kobe_output : pandas dataframe
```

Comment:
somehow it would be nice to see that this kobe shadows output. Using df_kobe_output suggests that this is the final output, already..